### PR TITLE
Define the converter API interface

### PR DESCRIPTION
**Pull request by gunjan5**
## Todos
- [x] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/upgrade/etcd/conversionv1v3/converter.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+)
+
+type Resource interface {
+	runtime.Object
+	v1.ObjectMetaAccessor
+}
+
+type Converter interface {
+	v1APITov1BackendConverter(unversioned.Resource) (*model.KVPair, error)
```
Comment (robbrockbank) [inline lib/upgrade/etcd/conversionv1v3/converter.go]: v1Backend -> V1Backend

File: lib/upgrade/etcd/conversionv1v3/converter.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	v1BackendTov3ResourceConverter(*model.KVPair) (Resource, error)
```
Comment (robbrockbank) [inline lib/upgrade/etcd/conversionv1v3/converter.go]: v3Resource -> V3Resource

File: lib/upgrade/etcd/conversionv1v3/converter.go (hunk 1 above)
Comment (robbrockbank) [reply lib/upgrade/etcd/conversionv1v3/converter.go]: for consistency with the v capitalization you could switch the ordering to apiV1ToBackendV1

but not too fussed

Comment (robbrockbank) [reply lib/upgrade/etcd/conversionv1v3/converter.go]: You can drop the "Converter" from the interface methods.

File: lib/upgrade/etcd/conversionv1v3/ippool.go [hunk 3]
```diff
@@ -21,31 +21,61 @@ import (
...
 	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
 	"github.com/projectcalico/libcalico-go/lib/ipip"
 	cnet "github.com/projectcalico/libcalico-go/lib/net"
 )
 
-func ConvertIPPool(kvp *model.KVPair) (*model.KVPair, error) {
-	pool, ok := kvp.Value.(model.IPPool)
+// ippool implements the Converter interface.
+type ippool struct{}
```
Comment (robbrockbank) [inline lib/upgrade/etcd/conversionv1v3/ippool.go]: Probably want to make this public, e.g. IPPoolConverter

Comment (robbrockbank) [reply lib/upgrade/etcd/conversionv1v3/ippool.go]: Oooh, actually I think you'll want the methods to be public too.  I suspect the upgrade code will not be written in this package.

Comment (gunjan5) [reply lib/upgrade/etcd/conversionv1v3/ippool.go]: yes good point, I'll export them

**Pull request merged**